File: leetsync/3490-find-the-maximum-length-of-valid-subsequence-i/find-the-maximum-length-of-valid-subsequence-i.java
class Solution {
    public int maximumLength(int[] nums) {
        int evenLength = 0, oddLength = 0;
        int n = nums.length;
        for (int num : nums) {
            if (num % 2 == 0) {
                evenLength++;
            } else {
                oddLength++;
            }
        }
        int altLength = 1; // At least 1 length
        int prevParity = nums[0] % 2;
        for (int i = 1; i < n; i++) {
            int currParity = nums[i] % 2;
            if (currParity != prevParity) {
                altLength++;
                prevParity = currParity;
            }
        }
        return Math.max(altLength, Math.max(evenLength, oddLength));
    }
}

File: leetsync/3490-find-the-maximum-length-of-valid-subsequence-i/find-the-maximum-length-of-valid-subsequence-i.go
func maximumLength(nums []int) int {
	n := len(nums)
	evenLen, oddLen := 0, 0
	for _, num := range nums {
		if num%2 == 0 {
			evenLen++
		} else {
			oddLen++
		}
	}
	altLen := 1
	prevParity := nums[0] % 2
	for i := 1; i < n; i++ {
		currParity := nums[i] % 2
		if currParity != prevParity {
			altLen++
			prevParity = currParity
		}
	}
	return max(altLen, evenLen, oddLen)
}